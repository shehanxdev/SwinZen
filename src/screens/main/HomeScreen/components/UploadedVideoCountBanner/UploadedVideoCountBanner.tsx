import React from 'react';
import { View } from 'react-native';

import { tw } from '@sz/config';
import { Color } from '@sz/constants';

import { VideoCountCard } from './components';

interface UploadedVideoCountBannerProps {
  videoCount: number; //refers to total uploaded video count from a specific month.
  swinZenUniVideoCount: number; //refers to total uploaded video count from all the times. Not just for a specific month.
}

export function UploadedVideoCountBanner({ videoCount, swinZenUniVideoCount }: UploadedVideoCountBannerProps) {
  return (
    <View style={[tw`flex flex-row justify-between min-h-15`]} testID="UploadedVideoCountBannerComponentTestID">
      <VideoCountCard count={videoCount} description={`Video\nUploads`} />
      <View style={tw`w-2.5`} />
      <VideoCountCard
        count={swinZenUniVideoCount}
        description={`SwingZen\nUniversity`}
        backgroundColor={Color.Tertiary.Sz900}
        countTextColor={Color.Neutral.Sz1000}
        descriptionTextColor={Color.Neutral.Sz1000}
      />
    </View>
  );
}
