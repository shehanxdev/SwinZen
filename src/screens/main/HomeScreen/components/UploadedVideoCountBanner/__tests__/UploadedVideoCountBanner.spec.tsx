import { render } from '@testing-library/react-native';
import * as React from 'react';

import { UploadedVideoCountBanner } from '../UploadedVideoCountBanner';

describe('UploadedVideoCountBanner Component', () => {
  const testID = 'UploadedVideoCountBannerComponentTestID';
  const videoCount = 24;
  const swinZenUniVideoCount = 82;

  const getRenderedComponent = () =>
    render(<UploadedVideoCountBanner videoCount={videoCount} swinZenUniVideoCount={swinZenUniVideoCount} />);

  it(`should render Component correctly`, () => {
    const renderer = getRenderedComponent();
    const renderTree = renderer.toJSON();
    expect(renderTree).toMatchSnapshot();
  });

  it('should find the Component via testID', () => {
    const { getByTestId } = getRenderedComponent();
    const foundScreen = getByTestId(testID);
    expect(foundScreen).toBeTruthy();
  });
});
