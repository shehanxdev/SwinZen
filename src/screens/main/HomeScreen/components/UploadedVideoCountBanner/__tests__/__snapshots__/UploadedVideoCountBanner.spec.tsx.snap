// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UploadedVideoCountBanner Component should render Component correctly 1`] = `
<View
  style={
    [
      {
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "minHeight": 60,
      },
    ]
  }
  testID="UploadedVideoCountBannerComponentTestID"
>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#1A5C23",
        "borderRadius": 10,
        "flexBasis": "0%",
        "flexDirection": "row",
        "flexGrow": 1,
        "flexShrink": 1,
        "justifyContent": "flex-start",
      }
    }
  >
    <View
      style={
        {
          "marginLeft": 10,
          "marginRight": 16,
        }
      }
    >
      <Text
        style={
          [
            {
              "color": "#FFFFFF",
              "lineHeight": 64,
            },
            {
              "fontFamily": "SourceSansPro-Regular",
              "fontSize": 48,
              "fontWeight": "bold",
              "textAlign": "center",
            },
          ]
        }
      >
        24
      </Text>
    </View>
    <Text
      style={
        [
          {
            "color": "#FFFFFF",
            "lineHeight": 24,
          },
          {
            "fontFamily": "SourceSansPro-Regular",
            "fontSize": 16,
            "fontWeight": "normal",
            "textAlign": "left",
          },
        ]
      }
    >
      Video
Uploads
    </Text>
  </View>
  <View
    style={
      {
        "width": 10,
      }
    }
  />
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#A2FD2F",
        "borderRadius": 10,
        "flexBasis": "0%",
        "flexDirection": "row",
        "flexGrow": 1,
        "flexShrink": 1,
        "justifyContent": "flex-start",
      }
    }
  >
    <View
      style={
        {
          "marginLeft": 10,
          "marginRight": 16,
        }
      }
    >
      <Text
        style={
          [
            {
              "color": "#283E36",
              "lineHeight": 64,
            },
            {
              "fontFamily": "SourceSansPro-Regular",
              "fontSize": 48,
              "fontWeight": "bold",
              "textAlign": "center",
            },
          ]
        }
      >
        82
      </Text>
    </View>
    <Text
      style={
        [
          {
            "color": "#283E36",
            "lineHeight": 24,
          },
          {
            "fontFamily": "SourceSansPro-Regular",
            "fontSize": 16,
            "fontWeight": "normal",
            "textAlign": "left",
          },
        ]
      }
    >
      SwingZen
University
    </Text>
  </View>
</View>
`;
